cpp/log/cert.cc

```diff
@@ -546,12 +566,314 @@ Cert::Status Cert::ExtensionStructure(int extension_nid,
       LOG(WARNING) << "Corrupt extension data";
       LOG_OPENSSL_ERRORS(WARNING);
     }
+
     return FALSE;
   }
 
   return TRUE;
 }
 
+bool Cert::IsRedactedHost(const std::string& hostname) {
+  // Split the hostname on '.' characters
+  std::vector<string> tokens = util::split(hostname, '.');
+
+  for (string str : tokens) {
+    if (str == "?") {
```

Comment:
Could this all be just something like:
`return hostname.find("?") != npos`
or
`return hostname.find(".?") != npos || hostname.find("?.") != npos || hostname == "?";`

Admittedly it's O(2N), but less code (or are you doing something slightly different to that?).